offer-manager-domain/src/main/groovy/life/qbic/business/customers/create/CreateCustomer.groovy

```diff
@@ -15,17 +16,24 @@ class CreateCustomer implements CreateCustomerInput {
 
   private CreateCustomerDataSource dataSource
   private CreateCustomerOutput output
+  private UpdateCustomer updateCustomer
 
 
   CreateCustomer(CreateCustomerOutput output, CreateCustomerDataSource dataSource){
     this.output = output
     this.dataSource = dataSource
+    this.updateCustomer = new UpdateCustomer(output,dataSource)
   }
 
   @Override
   void createCustomer(Customer customer) {
     try {
-      dataSource.addCustomer(customer)
+      int customerId = dataSource.findCustomer(customer).get()
```

Comment:
Bad news: If my aim is to change name or e-mail address of the customer, I won't find their ID here, leading to a duplicate entry. I think we absolutely need the customer identifier if we want to update customers.

Reply:
After a short discussion with @sven1103 we decided that introducing an ID is actually not required and we should differentiate in the use case by e.g introducing a new method that has two input parameters (oldCustomer, newCustomer). Good catch! 🚀

Reply:
That is also a possibility. Shall we make sure that the old customer is still active, i.e. that no other user has changed it in the meantime?